File: autumn/human-readable-number/dart/while.dart
String numberFormat(double d) {
   var e = d, f = 0;
   while (e >= 1000) {
      e /= 1000;
      f++;
   }
   return e.toStringAsFixed(3) + ['', ' k', ' M', ' G'][f];
}

void main() {
   var s = numberFormat(9012345678);
   print(s == '9.012 G');
}

File: autumn/human-readable-number/py/while.py
def number_format(d):
   e, f = d, 0
   while e >= 1e3:
      e /= 1e3
      f += 1
   return '%.3f' % e + ('', ' k', ' M', ' G')[f]

s = number_format(9012345678)
print(s == '9.012 G')
